google/cloud/storage/internal/rest/stub.cc

```diff
@@ -118,6 +118,20 @@ Status AddAuthorizationHeader(Options const& options,
   return {};
 }
 
+void AddCustomHeaders(Options const& options, RestRequestBuilder& builder) {
```

Comment:
I would probably have this function return a `Status` just like AddAuthorization header, then in `AddHeaders` do a similar status check.